src/main/java/com/google/opensesame/servlets/MentorsServlet.java

```diff
@@ -60,4 +74,55 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(jsonMentors);
   }
+
+  // This function prints an error message to the mentor form page if invalid input is found.
+  public void formError(HttpServletRequest request, HttpServletResponse response, String error)
+      throws ServletException, IOException {
+    String message =
+        "<ul><li class='ml-3 display-4 font-weight-bold text-danger'>" + error + "</li></ul>";
+    response.setContentType("text/html");
+    PrintWriter pw = response.getWriter();
+    pw.print(message);
+    RequestDispatcher rd = request.getRequestDispatcher("mentor_form.html");
+    rd.include(request, response);
+    pw.close();
+    return;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+    String repoUrl = request.getParameter("inputRepo");
+    String urlPattern = "https://github.com/";
+    Pattern regEx = Pattern.compile(urlPattern);
+    Matcher inputMatcher = regEx.matcher(repoUrl);
+    if (!inputMatcher.find()) {
+      // Input is not in the format of a GitHub repo link.
+      formError(request, response, "ENTER A VALID REPO LINK");
+      return;
+    }
```

Comment:
I think that it would be simpler to just rely on the `gitHub.getRepository` function to figure out whether a repository is valid or not. If a url doesn't start with `https://github.com/` I'd assume that the `gitHub.getRepository` function wouldn't be able to find the repo and then the user would get the `COULDN'T FIND REPO` message and they'd be able to figure out what was wrong from that error message.
```suggestion
```

Reply:
Good idea! Thank you. I guess using a regular expression was unecessary here.